power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/dataset.hpp

```diff
@@ -0,0 +1,170 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_DATASET_HPP
+#define POWER_GRID_MODEL_CPP_DATASET_HPP
+
+#include "power_grid_model_c/dataset.h"
+
+#include "basics.hpp"
+#include "handle.hpp"
+#include "serialization.hpp"
+
+namespace power_grid_model_cpp {
+class Info {
+  public:
+    Info(RawDatasetInfo const* info) : info_{info} {}
+
+    static std::string const dataset_name(Info const& info) {
+        auto const name = std::string(PGM_dataset_info_name(info.handle_.get(), info.info_.get()));
+        info.handle_.check_error();
+        return name;
+    }
+    std::string const dataset_name() const { return dataset_name(*this); }
+
+    static Idx dataset_is_batch(Info const& info) {
+        auto const is_batch = PGM_dataset_info_is_batch(info.handle_.get(), info.info_.get());
+        info.handle_.check_error();
+        return is_batch;
+    }
+    Idx dataset_is_batch() const { return dataset_is_batch(*this); }
+
+    static Idx dataset_batch_size(Info const& info) {
+        auto const batch_size = PGM_dataset_info_batch_size(info.handle_.get(), info.info_.get());
+        info.handle_.check_error();
+        return batch_size;
+    }
+    Idx dataset_batch_size() const { return dataset_batch_size(*this); }
+
+    static Idx dataset_n_components(Info const& info) {
+        auto const n_components = PGM_dataset_info_n_components(info.handle_.get(), info.info_.get());
+        info.handle_.check_error();
+        return n_components;
+    }
+    Idx dataset_n_components() const { return dataset_n_components(*this); }
+
+    static std::string const component_name(Info const& info, Idx component_idx) {
+        auto const component_name =
+            std::string(PGM_dataset_info_component_name(info.handle_.get(), info.info_.get(), component_idx));
```

Comment:
in general, it's better to use curly brace constructors (because there's a subtle but important difference between a constructor and a function)
```suggestion
        std::string component_name{PGM_dataset_info_component_name(info.handle_.get(), info.info_.get(), component_idx)};
```